clients/include/testing_rotm.hpp

```diff
@@ -0,0 +1,189 @@
+                if(arg.norm_check)
+                {
+                    norm_error_host_x = norm_check_general<T>('F', 1, N, incx, cx, rx);
+                    norm_error_host_y = norm_check_general<T>('F', 1, N, incy, cy, ry);
+                }
+            }
+
+            // Test rocblas_pointer_mode_device
+            {
+                CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+                CHECK_HIP_ERROR(hipMemcpy(dx, hx, sizeof(T) * size_x, hipMemcpyHostToDevice));
+                CHECK_HIP_ERROR(hipMemcpy(dy, hy, sizeof(T) * size_y, hipMemcpyHostToDevice));
+                CHECK_HIP_ERROR(hipMemcpy(dparam, hparam, sizeof(T) * 5, hipMemcpyHostToDevice));
+                CHECK_ROCBLAS_ERROR(rocblas_rotm<T>(handle, N, dx, incx, dy, incy, dparam));
+                host_vector<T> rx(size_x);
+                host_vector<T> ry(size_y);
+                CHECK_HIP_ERROR(hipMemcpy(rx, dx, sizeof(T) * size_x, hipMemcpyDeviceToHost));
+                CHECK_HIP_ERROR(hipMemcpy(ry, dy, sizeof(T) * size_y, hipMemcpyDeviceToHost));
+                if(arg.unit_check)
+                {
+                    unit_check_general<T>(1, N, incx, cx, rx);
+                    unit_check_general<T>(1, N, incy, cy, ry);
+                }
+                if(arg.norm_check)
+                {
+                    norm_error_device_x = norm_check_general<T>('F', 1, N, incx, cx, rx);
+                    norm_error_device_y = norm_check_general<T>('F', 1, N, incy, cy, ry);
+                }
+            }
+        }
+    }
+
+    if(arg.timing)
+    {
+        int number_cold_calls = 2;
+        int number_hot_calls  = 100;
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        CHECK_HIP_ERROR(hipMemcpy(dx, hx, sizeof(T) * size_x, hipMemcpyHostToDevice));
+        CHECK_HIP_ERROR(hipMemcpy(dy, hy, sizeof(T) * size_y, hipMemcpyHostToDevice));
+
+        for(int iter = 0; iter < number_cold_calls; iter++)
+        {
+            rocblas_rotm<T>(handle, N, dx, incx, dy, incy, hparam);
+        }
+        gpu_time_used = get_time_us(); // in microseconds
+        for(int iter = 0; iter < number_hot_calls; iter++)
+        {
+            rocblas_rotm<T>(handle, N, dx, incx, dy, incy, hparam);
+        }
+        gpu_time_used = (get_time_us() - gpu_time_used) / number_hot_calls;
+
+        std::cout << "N,incx,incy,rocblas(us),cpu(us)";
+        if(arg.norm_check)
+            std::cout
+                << ",norm_error_host_x,norm_error_host_y,norm_error_device_x,norm_error_device_y";
+        std::cout << std::endl;
+        std::cout << N << "," << incx << "," << incy << "," << gpu_time_used << ","
+                  << cpu_time_used;
+        if(arg.norm_check)
+            std::cout << ',' << norm_error_host_x << ',' << norm_error_host_y << ","
+                      << norm_error_device_x << "," << norm_error_device_y;
+        std::cout << std::endl;
+    }
+}
```

Comment:
The testing_rot* functions should be consolidated into one file, not duplicated.